### PR TITLE
PriceRegistry - remove token price staleness validations

### PR DESCRIPTION
**Pull request by elatoskinas**
## Motivation
The token price & gas price staleness validations in the PriceRegistry serve as an update staleness validation. However, since this is already captured by the gas price updates, the token price staleness checks can be removed completely, especially since most tokens will be priced through data feeds (where the staleness check should be skipped for decoupling)

## Solution
* Removes the staleness threshold check for all token price flows, keeping the check only on `getTokenAndGasPrices`
* Refactors PriceRegistry price updaters to use the `AuthorizedCallers` shared contract

### CODE REVIEW

File: contracts/src/v0.8/ccip/PriceRegistry.sol [hunk 1]
```diff
@@ -27,7 +27,6 @@ contract PriceRegistry is IPriceRegistry, OwnerIsCreator, ITypeAndVersion {
   error ChainNotSupported(uint64 chain);
```
Comment (elatoskinas) [inline contracts/src/v0.8/ccip/PriceRegistry.sol:28]: TODO: integrate [AuthorizedCallers](https://github.com/smartcontractkit/ccip/pull/985/files#diff-5de6378ab873aa44a6db71ba65efb8161aa344926224f3a7cd8540ae9f22d786R9) shared contract

File: contracts/.changeset/slimy-games-fetch.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@chainlink/contracts-ccip": minor
+---
+
```
Comment (RensR) [inline contracts/.changeset/slimy-games-fetch.md]: pls remove

File: core/services/ocr2/plugins/ccip/testhelpers/ccip_contracts.go [hunk 3]
```diff
@@ -39,7 +39,7 @@ import (
 	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/lock_release_token_pool"
 	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/maybe_revert_message_receiver"
 	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/mock_arm_contract"
-	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/price_registry"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/price_registry_1_2_0"
```
Comment (RensR) [inline core/services/ocr2/plugins/ccip/testhelpers/ccip_contracts.go:42]: good call 👍🏼

File: contracts/src/v0.8/ccip/PriceRegistry.sol [hunk 4]
```diff
@@ -86,23 +84,22 @@ contract PriceRegistry is IPriceRegistry, OwnerIsCreator, ITypeAndVersion {
...
     }
 
-    (uint224 price, uint32 timestamp) = _getTokenPriceFromDataFeed(priceFeedConfig);
-    return Internal.TimestampedPackedUint224({value: price, timestamp: timestamp});
+    return _getTokenPriceFromDataFeed(priceFeedConfig);
   }
 
-  // @inheritdoc IPriceRegistry
+  /// @inheritdoc IPriceRegistry
   function getValidatedTokenPrice(address token) external view override returns (uint224) {
```
Comment (RensR) [inline contracts/src/v0.8/ccip/PriceRegistry.sol:98]: getValidatedTokenPrice is never actually used, is it? Isn't it now pretty much the same function as getTokePrice anyway? Maybe good to look at what functions can be removed

Comment (elatoskinas) [reply contracts/src/v0.8/ccip/PriceRegistry.sol:98]: It is still being used in the OnRamp pricing flows, both by calling `getValidatedTokenPrice` and `convertTokenAmount` - the remaining validation is making sure the fee token is supported:

```
    // Token price must be set at least once
    if (tokenPrice.timestamp == 0 || tokenPrice.value == 0) revert TokenNotSupported(token);
```

**Review by RensR — APPROVED** [review]

**Review by connorwstein — APPROVED** [review]

**Review by matYang — APPROVED** [review]

**Pull request merged**